src/lib/decodeBySigHash.ts

```diff
@@ -34,89 +22,74 @@ export async function fetchAndDecodeWithCalldata(
   }
 }
 
+export async function decodeWithEventProps(
+  sigHash: string,
+  eventProps: EventProps,
+): Promise<DecodedEventResult[] | undefined> {
+  const response = await fetch4BytesBy.EventProps(sigHash);
+  if (response) {
+    const ifaces = parse4BytesResToIfaces(response, 'event');
+    return decode4BytesData(ifaces, eventProps, decodeEvent);
+  }
+}
+
 export function sigHashFromCalldata(calldata: string): string | undefined {
   const chunk = calldata.slice(0, 10);
   if (hexSchema.safeParse(chunk).success) {
     return chunk;
   }
 }
 
-export type FetchResult = {
-  id: number;
-  text_signature: string;
-  bytes_signature: string;
-  created_at: string;
-  hex_signature: string;
+export const fetch4BytesBy = {
+  EventProps: async (sigHash: string) => {
+    return fetch4Bytes(sigHash, 'event-signatures');
+  },
+  Signatures: async (sigHash: string) => {
+    return fetch4Bytes(sigHash, 'signatures');
+  },
 };
 
-// @internal
-// there are more types, but we don't need them for now
-export type HexSigType = 'signatures' | 'event-signatures';
-
-type Bytes4Cache = {
-  [sigType in HexSigType]: {
-    // undefined - not populated
-    // [] - no results
-    // [...] - results
-    [sig: string]: FetchResult[] | undefined;
-  };
-};
```

Comment:
Well, this is still needed if we want our optimistic fetch to work. now it's a regression. Unfortunately we don't have a test for it (my bad). Would be great to have it. And a great way to check mocks. Will look smth like:
```
click()
assert.calledOnce()
click()
assert.called0Times()
```